include/boost/geometry/algorithms/detail/closest_points/result.hpp

```diff
@@ -0,0 +1,98 @@
+// Boost.Geometry
+
+// Copyright (c) 2019, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_ALGORITHMS_DETAIL_CLOSEST_POINTS_RESULT_HPP
+#define BOOST_GEOMETRY_ALGORITHMS_DETAIL_CLOSEST_POINTS_RESULT_HPP
+
+//#include <boost/geometry/formulas/point_segment_distance.hpp>
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace closest_points
+{
+
+template <typename CT>
+class result
+{
+public :
+
+    result()
+        : distance(0)
+        , lon1(0)
+        , lat1(0)
+        , lon2(0)
+        , lat2(0)
+    {}
+
+    result(CT init)
+        : distance(init)
+        , lon1(0)
+        , lat1(0)
+        , lon2(0)
+        , lat2(0)
+    {}
+
+    CT distance;
+    CT lon1;
+    CT lat1;
+    CT lon2;
+    CT lat2;
+
+    template <typename T>
+    void operator=(T const& res)
+    {
+        distance = res.distance;
+        lon1 = res.lon1;
+        lat1 = res.lat1;
+        lon2 = res.lon2;
+        lat2 = res.lat2;
+    }
+
+    bool operator<(const result& rhs) const
+    {
+        return distance < rhs.distance;
+    }
+
+    bool operator==(const result& rhs) const
+    {
+        return distance == rhs.distance;
+    }
+
+    void swap()
```

Comment:
Typically a member function called like that would allow to swap 2 objects of type `result`. How about calling it `swap_points` or something?

Reply:
I agree, `swap_points` sounds fine